### PR TITLE
fixed typo in prod_builders for ios platform channels benchmark

### LINKED ISSUE #82081: iOS platform channel benchmarks aren't running
Platform channel benchmarks are working for android (https://flutter-flutter-perf.skia.org/e/?queries=test%3Dlinux_platform_channels_benchmarks), but not for iOS (https://flutter-flutter-perf.skia.org/e/?queries=test%3Dmac_ios_platform_channels_benchmarks)

**Flutter commit:** https://github.com/flutter/flutter/pull/81414

**ifra commit:** https://github.com/flutter/infra/pull/432

**issue:** https://github.com/flutter/flutter/issues/81559

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/82081

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/prod_builders.json [hunk 1]
```diff
@@ -1149,7 +1149,7 @@
     {
       "name": "Mac_ios platform_channels_benchmarks",
```
Comment (keyonghan) [inline dev/prod_builders.json:1150]: With required update in infra config side, here may need update as well.

File: dev/prod_builders.json [hunk 2]
```diff
@@ -1149,7 +1149,7 @@
...
       "repo": "flutter",
-      "task_name": "mac_ios_platform_channels_benchmarks",
+      "task_name": "mac_ios_platform_channels_benchmarks_ios",
```
Comment (keyonghan) [inline dev/prod_builders.json:1152]: I guess the task name also needs to be updated here: https://devops-console-oss.corp.google.com/flutter/infra/+/master:config/devicelab_config.star;l=355, which controls which test file to run.

Comment (gaaclarke) [reply dev/prod_builders.json:1152]: Done here: https://github.com/flutter/infra/pull/446  Let me know what order you'd like me to merge if you have a preference.  Thanks again @keyonghan

Comment (keyonghan) [reply dev/prod_builders.json:1152]: Sure. After the infra PR lands, we can land this PR with updated builder name and task name.

**Review by keyonghan — APPROVED** [review]
LGTM with nit.

File: dev/prod_builders.json (hunk 1 above)
Comment (gaaclarke) [reply dev/prod_builders.json]: done

**Pull request merged**